pgxn/neon/logical_replication_monitor.c

```diff
@@ -43,28 +57,38 @@ LsnDescComparator(const void *a, const void *b)
  * slots having lower restart_lsn should be dropped.
  */
 static XLogRecPtr
-get_num_snap_files_lsn_threshold(void)
+get_snapshots_cutoff_lsn(void)
 {
+#define SNAPDIR "pg_logical/snapshots"
+
 	DIR		   *dirdesc;
+	int			dirdesc_fd;
 	struct dirent *de;
-	char	   *snap_path = "pg_logical/snapshots/";
-	int			lsns_allocated = 1024;
-	int			lsns_num = 0;
-	XLogRecPtr *lsns;
-	XLogRecPtr	cutoff;
+	size_t		snapshot_index = 0;
+	SnapDesc   *snapshot_descriptors;
+	size_t		descriptors_allocated = 1024;
+	XLogRecPtr	cutoff = 0;
+	off_t		logicalsnapdir_size = 0;
+	const int	logical_replication_max_logicalsnapdir_size_bytes = logical_replication_max_logicalsnapdir_size * 1000;
 
-	if (logical_replication_max_snap_files < 0)
+	if (logical_replication_max_snap_files < 0 || logical_replication_max_logicalsnapdir_size < 0)
```

Comment:
It means that both criteria should be defined. If we want to limit just number of files or just their size - it will not be possible.